cmd/autoheal/object_template_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+)
+
+func TestNewObjectTemplateBuilder(t *testing.T) {
+	b := NewObjectTemplateBuilder()
+	if b.right != "}}" {
+		t.Errorf("Expected }} right delimiter got %+v", b.right)
+	}
+	if b.left != "{{" {
+		t.Errorf("Expected {{ left delimiter got %+v", b.left)
+	}
+}
+
+func TestDelimiters(t *testing.T) {
+	b := ObjectTemplateBuilder{}
+	b.Delimiters("left", "right")
+	if b.right != "right" {
+		t.Errorf("Expected right delimiter, got %+v", b.right)
+	}
+	if b.left != "left" {
+		t.Errorf("Expected left delimiter, got %+v", b.left)
+	}
+}
+
+func TestVariable(t *testing.T) {
+	b := ObjectTemplateBuilder{}
+	b.Variable("foo", "bar")
+	if b.variables["foo"] != "bar" {
+		t.Errorf("Expected variable to be set to foo, got %+v", b.variables["foo"])
+	}
+}
+
+func TestBuild(t *testing.T) {
+	template, err := NewObjectTemplateBuilder().
+		Delimiters("left", "right").
+		Variable("foo", "bar").
+		Build()
+	if err != nil {
+		t.Errorf("Error when constructing ObjectTemplate: %+v", err)
+	}
+	if template.right != "right" {
+		t.Errorf("Expected right delimiter, got %+v", template.right)
+	}
+	if template.left != "left" {
+		t.Errorf("Expected left delimiter, got %+v", template.left)
+	}
+	if template.variables["foo"] != "bar" {
+		t.Errorf("Expected variable to be set to foo, got %+v", template.variables["foo"])
+	}
+}
+
+type TemplateTestDataNested struct {
+	StructKey string
+}
+
+type TemplateTestData struct {
+	Map    map[string]string
+	Str    string
+	Struct TemplateTestDataNested
+}
+
+func TestProcess(t *testing.T) {
+	template, err := NewObjectTemplateBuilder().
```

Comment:
This is a very good example of the kind of test that I think we should be doing.